doc/userguide/output/eve/eve-json-output.rst

```diff
@@ -317,6 +323,32 @@ This example will cause each Suricata thread to write to its own "eve.json" file
 by adding a unique identifier to the filename.  For example, ``eve.7.json``.
 
 
+.. _output_eve_flush:
+
+Flushing file contents
+--~~~~~~~~~~~~~~~~~~~~
+
+By default, Suricata will call ``flush`` each time the file is written. ``flush-threshold``
+can be set to a byte-count to reduce the number of ``fflush`` calls. When non-zero, ``fflush`` is called
+when the bytes written to the file are equal to or exceed the value; subsequent calls to ``fflush`` occur.
+
+Reducing the number of ``fflush`` calls can reduce write overhead at the expense of risk of leaving
+up to ``flush-threshold`` bytes unbuffered/unwritten should Suricata unexpectedly terminate.
+
+The default value is ``0``.
+
+::
+
+   outputs:
+     - eve-log:
+         filename: eve.json
+         flush-threshold: 4096
```

Comment:
I'd probably make this example value much higher.  Quite a lot of events are 4-8k.